cloud-reactor/src/main/java/com/sequenceiq/cloudbreak/cloud/handler/StopStartDownscaleStopInstancesHandler.java

```diff
@@ -0,0 +1,90 @@
+
+import com.sequenceiq.cloudbreak.cloud.CloudConnector;
+import com.sequenceiq.cloudbreak.cloud.context.AuthenticatedContext;
+import com.sequenceiq.cloudbreak.cloud.context.CloudContext;
+import com.sequenceiq.cloudbreak.cloud.init.CloudPlatformConnectors;
+import com.sequenceiq.cloudbreak.cloud.model.CloudInstance;
+import com.sequenceiq.cloudbreak.cloud.model.CloudVmInstanceStatus;
+import com.sequenceiq.cloudbreak.cloud.event.instance.StopStartDownscaleStopInstancesRequest;
+import com.sequenceiq.cloudbreak.cloud.event.instance.StopStartDownscaleStopInstancesResult;
+
+import reactor.bus.Event;
+import reactor.bus.EventBus;
+
+@Component
+public class StopStartDownscaleStopInstancesHandler implements CloudPlatformEventHandler<StopStartDownscaleStopInstancesRequest> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StopStartDownscaleStopInstancesHandler.class);
+
+    private static final Long STOP_POLL_TIMEBOUND_MS = 300_000L;
+
+    @Inject
+    private CloudPlatformConnectors cloudPlatformConnectors;
+
+    @Inject
+    private EventBus eventBus;
+
+    @Override
+    public Class<StopStartDownscaleStopInstancesRequest> type() {
+        return StopStartDownscaleStopInstancesRequest.class;
+    }
+
+    @Override
+    public void accept(Event<StopStartDownscaleStopInstancesRequest> event) {
+        StopStartDownscaleStopInstancesRequest request = event.getData();
+        LOGGER.info("StopStartDownscaleStopInstancesHandler: {}", event.getData().getResourceId());
+
+        CloudContext cloudContext = request.getCloudContext();
+        try {
+            CloudConnector<?> connector = cloudPlatformConnectors.get(cloudContext.getPlatformVariant());
+            AuthenticatedContext ac = getAuthenticatedContext(request, cloudContext, connector);
+
+            List<CloudInstance> cloudInstancesToStop = request.getCloudInstancesToStop();
+            List<CloudVmInstanceStatus> cloudVmInstanceStatusList = Collections.emptyList();
+            if (cloudInstancesToStop.size() > 0) {
+                LOGGER.info("Attempting to stop instances with a timebound of {}ms. count={}, instances=[{}]",
+                        STOP_POLL_TIMEBOUND_MS, cloudInstancesToStop.size(), cloudInstancesToStop.stream().map(CloudInstance::getInstanceId)
+                                .collect(Collectors.toList()));
+                // TODO CB-14929: Error handlnig. Currently - an error from the following command (e.g. NPE)
+                //  results in no WARNINGS on the console. The stop is just ignored, and the cluster goes into a
+                //  RUNNING state, with the nodes in a DECOMMISSIONED state. The error handling needs to make sure
+                //  that the system goes into a failed state, and there is at least some logging on the console
+                //  to indicate what happened.
+                // TODO CB-14929: Also, after re-provisioning nodes via CM, the CB UI does not seem to be refreshing the node
```

Comment:
I don't know what is reprovisioning via CM, but if you created a STUCK state (I don't reach that code piece yet if yes), then you should modify our stack status checker.

Reply:
The StackStatusHandler doesn't actually update states in CB based on what CM thinks. There's a thread about this on the internal slack channel. 'STUCK' is the wrong word to use here ... can re-word this to say that CM states are not updated by StackStatusHandler. Hunting through jiras, there's a jira filed which says the same as well.